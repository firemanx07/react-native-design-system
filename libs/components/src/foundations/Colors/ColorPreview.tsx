import { styled } from '@proxym/themes';
import React from 'react';

import { BaseText, TextVariant, TextWeight } from '../../atoms/BaseText';

export type PropsType = {
  color: string;
  textColor: string;
  name: string;
};

export const ColorPreview = ({ color, textColor, name }: PropsType) => {
  return (
    <Container>
      <ColorTile color={color}>
        <ColorText
          color={textColor}
          variant={TextVariant.M}
          weight={TextWeight.Bold}
        >
          {color}
        </ColorText>
      </ColorTile>
      <BaseText variant={TextVariant.M}>{name}</BaseText>
    </Container>
  );
};

const Container = styled.View`
  margin-right: 20px;
  margin-bottom: 20px;
`;

type CommonStyledType = {
  color?: string;
};

const ColorTile = styled.View<CommonStyledType>`
  background-color: ${({ color }) => color};
  border-radius: 8px;
  width: 100px;
  height: 50px;
  margin-bottom: 8px;
  justify-content: center;
  align-items: center;
  border-width: 1px;
  border-color: ${({ theme }) => theme.ds.colors.graySoft};
`;

const ColorText = styled(BaseText)<CommonStyledType>`
  color: ${({ color }) => color};
`;

export default ColorPreview;
