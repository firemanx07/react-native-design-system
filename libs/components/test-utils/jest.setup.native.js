import { Animated } from 'react-native';

require('react-native-reanimated/lib/reanimated2/jestUtils').setUpTests();

console.warn = () => {};

Animated.timing = () => ({
  start: () => jest.fn(),
});
