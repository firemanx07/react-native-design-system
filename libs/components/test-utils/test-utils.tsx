import { theme, ThemeProvider } from '@celebrate-app/themes';
import { Story } from '@storybook/react';
import {
  cleanup,
  render,
  RenderOptions,
  waitFor,
} from '@testing-library/react-native';
import React, { ReactNode } from 'react';

const Providers = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions
) => {
  return render(ui, { wrapper: Providers, ...options });
};

beforeEach(cleanup);
afterEach(cleanup);

export function testStory(displayName: string, TestComponent: Story) {
  test(`${displayName} snapshot test`, async () => {
    const comp = customRender(<TestComponent {...TestComponent.args} />);
    jest.runOnlyPendingTimers();

    await waitFor(() => {
      expect(comp).toBeTruthy();
      expect(comp.toJSON()).toMatchSnapshot();
    });
  });
}

export function testComponentSnapshot(
  TestComponent: React.ReactElement<
    any,
    string | React.JSXElementConstructor<any>
  >
) {
  test('Snapshot test', () => {
    const comp = customRender(TestComponent);
    expect(comp).toBeTruthy();
    expect(comp.toJSON()).toMatchSnapshot();
  });
}

export function testAllStories(stories: any) {
  Object.keys(stories).forEach((story) => {
    if (story !== 'default') {
      testStory(story, stories[story]);
    }
  });
}
