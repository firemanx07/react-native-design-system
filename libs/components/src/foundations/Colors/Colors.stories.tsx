import { Story } from '@storybook/react';
import React from 'react';

import { Colors as ColorsBase } from './Colors';

export default {
  component: ColorsBase,
  title: 'Foundations/Colors',
};

const Template: Story = () => <ColorsBase />;

const parameters = {
  controls: {
    include: [],
  },
};

const defaultArgs = {};

export const Colors = Template.bind({});

Colors.parameters = parameters;
Colors.args = {
  ...defaultArgs,
};
