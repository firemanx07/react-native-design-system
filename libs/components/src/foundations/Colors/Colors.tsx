import { styled, useNamespacedTheme } from '@proxym/themes';
import React, { Fragment, useCallback } from 'react';

import { BaseText, TextVariant } from '../../atoms/BaseText';
import ColorPreview, { PropsType as ColorPreviewProps } from './ColorPreview';

type SectionItem = {
  title: string;
  data: ColorPreviewProps[];
};

export const Colors = () => {
  const { colors } = useNamespacedTheme();

  const sections: SectionItem[] = [
    {
      title: 'Primary colors:',
      data: [
        {
          name: 'primary',
          color: colors.primary,
          textColor: colors.light,
        },
        {
          name: 'dark',
          color: colors.dark,
          textColor: colors.light,
        },
        {
          name: 'light',
          color: colors.light,
          textColor: colors.dark,
        },
      ],
    },
    {
      title: 'Secondary colors:',
      data: [
        {
          name: 'secondary',
          color: colors.secondary,
          textColor: colors.dark,
        },
        {
          name: 'secondaryDark',
          color: colors.secondaryDark,
          textColor: colors.dark,
        },
      ],
    },
    {
      title: 'Alert colors:',
      data: [
        {
          name: 'error',
          color: colors.error,
          textColor: colors.light,
        },
        {
          name: 'errorLight',
          color: colors.errorLight,
          textColor: colors.dark,
        },
        {
          name: 'success',
          color: colors.success,
          textColor: colors.light,
        },
        {
          name: 'successLight',
          color: colors.successLight,
          textColor: colors.dark,
        },
      ],
    },
    {
      title: 'Accent colors:',
      data: [
        {
          name: 'accent',
          color: colors.accent,
          textColor: colors.light,
        },
        {
          name: 'accentDark',
          color: colors.accentDark,
          textColor: colors.light,
        },
      ],
    },
    {
      title: 'Grayscale colors:',
      data: [
        {
          name: 'grayDark',
          color: colors.grayDark,
          textColor: colors.light,
        },
        {
          name: 'graySoft',
          color: colors.graySoft,
          textColor: colors.dark,
        },
        {
          name: 'grayLight',
          color: colors.grayLight,
          textColor: colors.dark,
        },
        {
          name: 'grayUltraLight',
          color: colors.grayUltraLight,
          textColor: colors.dark,
        },
      ],
    },
  ];

  const renderColor = useCallback(
    (item: ColorPreviewProps) => <ColorPreview key={item.name} {...item} />,
    [],
  );

  const renderSection = useCallback(
    (item: SectionItem) => (
      <Fragment key={item.title}>
        <BaseText variant={TextVariant.H2}>{item.title}</BaseText>
        <Section>{item.data.map(renderColor)}</Section>
      </Fragment>
    ),
    [],
  );

  return <Container>{sections.map(renderSection)}</Container>;
};

const Container = styled.View`
  padding: 20px;
`;

const Section = styled.View`
  width: 720px;
  flex-direction: row;
  flex-wrap: wrap;
  margin-bottom: 20px;
  margin-top: 10px;
`;

export default Colors;
