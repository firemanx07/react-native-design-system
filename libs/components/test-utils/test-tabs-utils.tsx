import { waitFor } from '@testing-library/react-native';

import { customRender } from './test-utils';

type renderTabsProps = {
  elem: JSX.Element;
  testIDs: string[];
};
export const renderTabs = ({ elem, testIDs }: renderTabsProps) => {
  const comp = customRender(elem);
  const tabs: ReturnType<typeof comp.getByTestId>[] = [];
  testIDs.forEach(testID => tabs.push(comp.getByTestId(testID)));
  return { comp, tabs };
};
export const testTabs = async (args: ReturnType<typeof renderTabs>) => {
  await waitFor(() => {
    expect(args.comp).toBeTruthy();
    args.tabs.forEach(tab => expect(tab).toBeDefined());
  });
};
