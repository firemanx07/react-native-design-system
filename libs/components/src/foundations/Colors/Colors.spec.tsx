import { testAllStories } from '../../../test-utils';
import * as stories from './Colors.stories';

describe('Colors tests', () => {
  testAllStories(stories);
});
