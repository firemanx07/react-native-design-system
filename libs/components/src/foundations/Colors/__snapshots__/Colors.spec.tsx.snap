// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colors tests Colors snapshot test 1`] = `
<View
  style={
    [
      {
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
      },
    ]
  }
>
  <Text
    fontFamily="Muli-Bold"
    fontSize={18}
    lineHeight={24}
    style={
      [
        {
          "color": "#222222",
          "fontFamily": "Muli-Bold",
          "fontSize": 18,
          "lineHeight": 24,
        },
      ]
    }
  >
    Primary colors:
  </Text>
  <View
    style={
      [
        {
          "flexDirection": "row",
          "flexWrap": "wrap",
          "marginBottom": 20,
          "marginTop": 10,
          "width": 720,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "marginBottom": 20,
            "marginRight": 20,
          },
        ]
      }
    >
      <View
        color="#FF8066"
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#FF8066",
              "borderColor": "#DEDFE0",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 8,
              "width": 100,
            },
          ]
        }
      >
        <Text
          color="#FFFFFF"
          fontFamily="Muli-Bold"
          fontSize={14}
          lineHeight={20}
          style={
            [
              {
                "color": "#222222",
                "fontFamily": "Muli-Bold",
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          #FF8066
        </Text>
      </View>
      <Text
        fontFamily="Muli-Regular"
        fontSize={14}
        lineHeight={20}
        style={
          [
            {
              "color": "#222222",
              "fontFamily": "Muli-Regular",
              "fontSize": 14,
              "lineHeight": 20,
            },
          ]
        }
      >
        primary
      </Text>
    </View>
    <View
      style={
        [
          {
            "marginBottom": 20,
            "marginRight": 20,
          },
        ]
      }
    >
      <View
        color="#222222"
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#222222",
              "borderColor": "#DEDFE0",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 8,
              "width": 100,
            },
          ]
        }
      >
        <Text
          color="#FFFFFF"
          fontFamily="Muli-Bold"
          fontSize={14}
          lineHeight={20}
          style={
            [
              {
                "color": "#222222",
                "fontFamily": "Muli-Bold",
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          #222222
        </Text>
      </View>
      <Text
        fontFamily="Muli-Regular"
        fontSize={14}
        lineHeight={20}
        style={
          [
            {
              "color": "#222222",
              "fontFamily": "Muli-Regular",
              "fontSize": 14,
              "lineHeight": 20,
            },
          ]
        }
      >
        dark
      </Text>
    </View>
    <View
      style={
        [
          {
            "marginBottom": 20,
            "marginRight": 20,
          },
        ]
      }
    >
      <View
        color="#FFFFFF"
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": "#DEDFE0",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 8,
              "width": 100,
            },
          ]
        }
      >
        <Text
          color="#222222"
          fontFamily="Muli-Bold"
          fontSize={14}
          lineHeight={20}
          style={
            [
              {
                "color": "#222222",
                "fontFamily": "Muli-Bold",
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "color": "#222222",
              },
            ]
          }
        >
          #FFFFFF
        </Text>
      </View>
      <Text
        fontFamily="Muli-Regular"
        fontSize={14}
        lineHeight={20}
        style={
          [
            {
              "color": "#222222",
              "fontFamily": "Muli-Regular",
              "fontSize": 14,
              "lineHeight": 20,
            },
          ]
        }
      >
        light
      </Text>
    </View>
  </View>
  <Text
    fontFamily="Muli-Bold"
    fontSize={18}
    lineHeight={24}
    style={
      [
        {
          "color": "#222222",
          "fontFamily": "Muli-Bold",
          "fontSize": 18,
          "lineHeight": 24,
        },
      ]
    }
  >
    Secondary colors:
  </Text>
  <View
    style={
      [
        {
          "flexDirection": "row",
          "flexWrap": "wrap",
          "marginBottom": 20,
          "marginTop": 10,
          "width": 720,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "marginBottom": 20,
            "marginRight": 20,
          },
        ]
      }
    >
      <View
        color="#FFF1E7"
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#FFF1E7",
              "borderColor": "#DEDFE0",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 8,
              "width": 100,
            },
          ]
        }
      >
        <Text
          color="#222222"
          fontFamily="Muli-Bold"
          fontSize={14}
          lineHeight={20}
          style={
            [
              {
                "color": "#222222",
                "fontFamily": "Muli-Bold",
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "color": "#222222",
              },
            ]
          }
        >
          #FFF1E7
        </Text>
      </View>
      <Text
        fontFamily="Muli-Regular"
        fontSize={14}
        lineHeight={20}
        style={
          [
            {
              "color": "#222222",
              "fontFamily": "Muli-Regular",
              "fontSize": 14,
              "lineHeight": 20,
            },
          ]
        }
      >
        secondary
      </Text>
    </View>
    <View
      style={
        [
          {
            "marginBottom": 20,
            "marginRight": 20,
          },
        ]
      }
    >
      <View
        color="#FFE6D1"
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#FFE6D1",
              "borderColor": "#DEDFE0",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 8,
              "width": 100,
            },
          ]
        }
      >
        <Text
          color="#222222"
          fontFamily="Muli-Bold"
          fontSize={14}
          lineHeight={20}
          style={
            [
              {
                "color": "#222222",
                "fontFamily": "Muli-Bold",
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "color": "#222222",
              },
            ]
          }
        >
          #FFE6D1
        </Text>
      </View>
      <Text
        fontFamily="Muli-Regular"
        fontSize={14}
        lineHeight={20}
        style={
          [
            {
              "color": "#222222",
              "fontFamily": "Muli-Regular",
              "fontSize": 14,
              "lineHeight": 20,
            },
          ]
        }
      >
        secondaryDark
      </Text>
    </View>
  </View>
  <Text
    fontFamily="Muli-Bold"
    fontSize={18}
    lineHeight={24}
    style={
      [
        {
          "color": "#222222",
          "fontFamily": "Muli-Bold",
          "fontSize": 18,
          "lineHeight": 24,
        },
      ]
    }
  >
    Alert colors:
  </Text>
  <View
    style={
      [
        {
          "flexDirection": "row",
          "flexWrap": "wrap",
          "marginBottom": 20,
          "marginTop": 10,
          "width": 720,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "marginBottom": 20,
            "marginRight": 20,
          },
        ]
      }
    >
      <View
        color="#F54F69"
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#F54F69",
              "borderColor": "#DEDFE0",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 8,
              "width": 100,
            },
          ]
        }
      >
        <Text
          color="#FFFFFF"
          fontFamily="Muli-Bold"
          fontSize={14}
          lineHeight={20}
          style={
            [
              {
                "color": "#222222",
                "fontFamily": "Muli-Bold",
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          #F54F69
        </Text>
      </View>
      <Text
        fontFamily="Muli-Regular"
        fontSize={14}
        lineHeight={20}
        style={
          [
            {
              "color": "#222222",
              "fontFamily": "Muli-Regular",
              "fontSize": 14,
              "lineHeight": 20,
            },
          ]
        }
      >
        error
      </Text>
    </View>
    <View
      style={
        [
          {
            "marginBottom": 20,
            "marginRight": 20,
          },
        ]
      }
    >
      <View
        color="#FEEDF0"
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#FEEDF0",
              "borderColor": "#DEDFE0",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 8,
              "width": 100,
            },
          ]
        }
      >
        <Text
          color="#222222"
          fontFamily="Muli-Bold"
          fontSize={14}
          lineHeight={20}
          style={
            [
              {
                "color": "#222222",
                "fontFamily": "Muli-Bold",
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "color": "#222222",
              },
            ]
          }
        >
          #FEEDF0
        </Text>
      </View>
      <Text
        fontFamily="Muli-Regular"
        fontSize={14}
        lineHeight={20}
        style={
          [
            {
              "color": "#222222",
              "fontFamily": "Muli-Regular",
              "fontSize": 14,
              "lineHeight": 20,
            },
          ]
        }
      >
        errorLight
      </Text>
    </View>
    <View
      style={
        [
          {
            "marginBottom": 20,
            "marginRight": 20,
          },
        ]
      }
    >
      <View
        color="#2FBFA7"
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#2FBFA7",
              "borderColor": "#DEDFE0",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 8,
              "width": 100,
            },
          ]
        }
      >
        <Text
          color="#FFFFFF"
          fontFamily="Muli-Bold"
          fontSize={14}
          lineHeight={20}
          style={
            [
              {
                "color": "#222222",
                "fontFamily": "Muli-Bold",
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          #2FBFA7
        </Text>
      </View>
      <Text
        fontFamily="Muli-Regular"
        fontSize={14}
        lineHeight={20}
        style={
          [
            {
              "color": "#222222",
              "fontFamily": "Muli-Regular",
              "fontSize": 14,
              "lineHeight": 20,
            },
          ]
        }
      >
        success
      </Text>
    </View>
    <View
      style={
        [
          {
            "marginBottom": 20,
            "marginRight": 20,
          },
        ]
      }
    >
      <View
        color="#EAF9F6"
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#EAF9F6",
              "borderColor": "#DEDFE0",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 8,
              "width": 100,
            },
          ]
        }
      >
        <Text
          color="#222222"
          fontFamily="Muli-Bold"
          fontSize={14}
          lineHeight={20}
          style={
            [
              {
                "color": "#222222",
                "fontFamily": "Muli-Bold",
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "color": "#222222",
              },
            ]
          }
        >
          #EAF9F6
        </Text>
      </View>
      <Text
        fontFamily="Muli-Regular"
        fontSize={14}
        lineHeight={20}
        style={
          [
            {
              "color": "#222222",
              "fontFamily": "Muli-Regular",
              "fontSize": 14,
              "lineHeight": 20,
            },
          ]
        }
      >
        successLight
      </Text>
    </View>
  </View>
  <Text
    fontFamily="Muli-Bold"
    fontSize={18}
    lineHeight={24}
    style={
      [
        {
          "color": "#222222",
          "fontFamily": "Muli-Bold",
          "fontSize": 18,
          "lineHeight": 24,
        },
      ]
    }
  >
    Accent colors:
  </Text>
  <View
    style={
      [
        {
          "flexDirection": "row",
          "flexWrap": "wrap",
          "marginBottom": 20,
          "marginTop": 10,
          "width": 720,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "marginBottom": 20,
            "marginRight": 20,
          },
        ]
      }
    >
      <View
        color="#FFB500"
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#FFB500",
              "borderColor": "#DEDFE0",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 8,
              "width": 100,
            },
          ]
        }
      >
        <Text
          color="#FFFFFF"
          fontFamily="Muli-Bold"
          fontSize={14}
          lineHeight={20}
          style={
            [
              {
                "color": "#222222",
                "fontFamily": "Muli-Bold",
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          #FFB500
        </Text>
      </View>
      <Text
        fontFamily="Muli-Regular"
        fontSize={14}
        lineHeight={20}
        style={
          [
            {
              "color": "#222222",
              "fontFamily": "Muli-Regular",
              "fontSize": 14,
              "lineHeight": 20,
            },
          ]
        }
      >
        accent
      </Text>
    </View>
    <View
      style={
        [
          {
            "marginBottom": 20,
            "marginRight": 20,
          },
        ]
      }
    >
      <View
        color="#FF9C48"
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#FF9C48",
              "borderColor": "#DEDFE0",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 8,
              "width": 100,
            },
          ]
        }
      >
        <Text
          color="#FFFFFF"
          fontFamily="Muli-Bold"
          fontSize={14}
          lineHeight={20}
          style={
            [
              {
                "color": "#222222",
                "fontFamily": "Muli-Bold",
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          #FF9C48
        </Text>
      </View>
      <Text
        fontFamily="Muli-Regular"
        fontSize={14}
        lineHeight={20}
        style={
          [
            {
              "color": "#222222",
              "fontFamily": "Muli-Regular",
              "fontSize": 14,
              "lineHeight": 20,
            },
          ]
        }
      >
        accentDark
      </Text>
    </View>
  </View>
  <Text
    fontFamily="Muli-Bold"
    fontSize={18}
    lineHeight={24}
    style={
      [
        {
          "color": "#222222",
          "fontFamily": "Muli-Bold",
          "fontSize": 18,
          "lineHeight": 24,
        },
      ]
    }
  >
    Grayscale colors:
  </Text>
  <View
    style={
      [
        {
          "flexDirection": "row",
          "flexWrap": "wrap",
          "marginBottom": 20,
          "marginTop": 10,
          "width": 720,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "marginBottom": 20,
            "marginRight": 20,
          },
        ]
      }
    >
      <View
        color="#666666"
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#666666",
              "borderColor": "#DEDFE0",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 8,
              "width": 100,
            },
          ]
        }
      >
        <Text
          color="#FFFFFF"
          fontFamily="Muli-Bold"
          fontSize={14}
          lineHeight={20}
          style={
            [
              {
                "color": "#222222",
                "fontFamily": "Muli-Bold",
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          #666666
        </Text>
      </View>
      <Text
        fontFamily="Muli-Regular"
        fontSize={14}
        lineHeight={20}
        style={
          [
            {
              "color": "#222222",
              "fontFamily": "Muli-Regular",
              "fontSize": 14,
              "lineHeight": 20,
            },
          ]
        }
      >
        grayDark
      </Text>
    </View>
    <View
      style={
        [
          {
            "marginBottom": 20,
            "marginRight": 20,
          },
        ]
      }
    >
      <View
        color="#DEDFE0"
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#DEDFE0",
              "borderColor": "#DEDFE0",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 8,
              "width": 100,
            },
          ]
        }
      >
        <Text
          color="#222222"
          fontFamily="Muli-Bold"
          fontSize={14}
          lineHeight={20}
          style={
            [
              {
                "color": "#222222",
                "fontFamily": "Muli-Bold",
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "color": "#222222",
              },
            ]
          }
        >
          #DEDFE0
        </Text>
      </View>
      <Text
        fontFamily="Muli-Regular"
        fontSize={14}
        lineHeight={20}
        style={
          [
            {
              "color": "#222222",
              "fontFamily": "Muli-Regular",
              "fontSize": 14,
              "lineHeight": 20,
            },
          ]
        }
      >
        graySoft
      </Text>
    </View>
    <View
      style={
        [
          {
            "marginBottom": 20,
            "marginRight": 20,
          },
        ]
      }
    >
      <View
        color="#F2F3F4"
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#F2F3F4",
              "borderColor": "#DEDFE0",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 8,
              "width": 100,
            },
          ]
        }
      >
        <Text
          color="#222222"
          fontFamily="Muli-Bold"
          fontSize={14}
          lineHeight={20}
          style={
            [
              {
                "color": "#222222",
                "fontFamily": "Muli-Bold",
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "color": "#222222",
              },
            ]
          }
        >
          #F2F3F4
        </Text>
      </View>
      <Text
        fontFamily="Muli-Regular"
        fontSize={14}
        lineHeight={20}
        style={
          [
            {
              "color": "#222222",
              "fontFamily": "Muli-Regular",
              "fontSize": 14,
              "lineHeight": 20,
            },
          ]
        }
      >
        grayLight
      </Text>
    </View>
    <View
      style={
        [
          {
            "marginBottom": 20,
            "marginRight": 20,
          },
        ]
      }
    >
      <View
        color="#F9F9FA"
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#F9F9FA",
              "borderColor": "#DEDFE0",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 8,
              "width": 100,
            },
          ]
        }
      >
        <Text
          color="#222222"
          fontFamily="Muli-Bold"
          fontSize={14}
          lineHeight={20}
          style={
            [
              {
                "color": "#222222",
                "fontFamily": "Muli-Bold",
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "color": "#222222",
              },
            ]
          }
        >
          #F9F9FA
        </Text>
      </View>
      <Text
        fontFamily="Muli-Regular"
        fontSize={14}
        lineHeight={20}
        style={
          [
            {
              "color": "#222222",
              "fontFamily": "Muli-Regular",
              "fontSize": 14,
              "lineHeight": 20,
            },
          ]
        }
      >
        grayUltraLight
      </Text>
    </View>
  </View>
</View>
`;
